src/components/date-picker/dropdown.tsx

```diff
@@ -0,0 +1,131 @@
+import * as React from 'react';
+import {getMonthNames, getMonthFromOffset, getDayNames, getDaysInMonth, getNumOfPreviousDays, getNumOfFollowingDays} from '../../common/date-helpers';
+import {observable, action, computed} from 'mobx';
+import {observer} from 'mobx-react';
+import {Day} from './day';
+const styles = require('./date-picker.st.css').default;
+
+export interface DropdownProps {
+    value: Date;
+    onChange(date: Date): void;
+}
+
+const monthNames = getMonthNames();
+
+@observer
+export class Dropdown extends React.Component<DropdownProps, {}>{
```

Comment:
I would separate everything related to calendar grid to `<Grid/>` component, so when we will have `<Dropdown/>` or `<Popup />` component we can use `<Grid/>` here and it will be less painful refactoring.